### ISSUE TITLE
VS 16.3.0 Preview 2 - IDE0060 Double Underscore False Positive

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#38381]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/716180/vs-1630-preview-2-ide0060-double-underscore-false-1.html)._

---
IDE0060 Remove unused parameter '__' is thrown here on the double underscore. Shouldn't this be considered a valid pattern for a discard variable?

```
namespace IDE0060_Double_Underscore
{
    public class C
    {
        string M(int _, int __)
        {
            return string. Empty;
        }
    }
}
```

---
#### Visual Studio Feedback System on 8/28/2019, 09:22 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by vatsalyaagrawal** [comment]
Is this a duplicate?

**Comment by mavasani** [comment]
This is by design. We use the following heuristic for discard symbol names:
https://github.com/dotnet/roslyn/blob/65a0d2ba1580d79ab9397981bf99aa823b25a491/src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs#L328-L335

The reported diagnostic also has a corresponding description on how to mark a parameter as discard:

![image](https://user-images.githubusercontent.com/10605811/64211512-4b2b8580-ce5b-11e9-8dfe-5a81fd501ec7.png)
